### PR TITLE
AP792 Delegated confirmation page

### PR DESCRIPTION
**Pull request by Obsiye**
Add a delegated functions confirmation page after the substantive page when using delegated functions and clicking 'no, i'll start later' on the substantive page.

[Link to story](https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=233&projectKey=AP&modal=detail&selectedIssue=AP-972)

- Add a controller, view, locales, unit tests and feature tests
- Update the provider flow
- Update existing tests

## Checklist

Before you ask people to review this PR:

- [x] Tests and rubocop should be passing: `bundle exec rake`
- [X] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [X] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [X] The PR description should say what you changed and why, with a link to the JIRA story.
- [X] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [X] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: config/locales/en/providers.yml [hunk 1]
```diff
@@ -129,6 +129,15 @@ en:
...
       show:
         h1-heading: Latest incident details
+    delegated_confirmation:
+      index:
+        page-heading: You notified us you've used delegated functions
+        sub_title: Your case reference number is
+        what_happens_next: What happens next
+        info: You can return to this application when you're ready to make a substantive application.
+        warning:
+          application_deadline_date: You must submit a substantive application by 10 November 2019.
```
Comment (mpw5) [inline config/locales/en/providers.yml]: I don't think the date should be hardcoded here - won't it vary from case to case?

File: config/routes.rb [hunk 2]
```diff
@@ -198,6 +198,7 @@
       resource :substantive_application, only: %i[show update]
       resource :end_of_application, only: %i[show update]
       resource :submitted_application, only: :show
+      resources :delegated_confirmation, only: :index
```
Comment (mpw5) [inline config/routes.rb]: Should this be `resource` not `resources`?

File: config/locales/en/providers.yml (hunk 1 above)
Comment (Obsiye) [reply config/locales/en/providers.yml]: 🤕 yes, bad mistake.

File: config/locales/en/providers.yml [hunk 3]
```diff
@@ -136,7 +136,7 @@ en:
         what_happens_next: What happens next
         info: You can return to this application when you're ready to make a substantive application.
         warning:
-          application_deadline_date: You must submit a substantive application by 10 November 2019.
+          application_deadline_date: "You must submit a substantive application by "
```
Comment (mpw5) [inline config/locales/en/providers.yml]: Are the `"`s necessary (Genuine question, I don't know! It looks like some translations use them, some don't. Is it for interpolation?)

**Pull request merged**